src/services/qix/session.js

```diff
@@ -99,6 +96,42 @@ class Session {
     this.emit('session-created', this);
   }
 
+  // TODO: Rename as this does not register the notification listerners.
+  /**
+  * Function used register the RPC listerners except for the notification listeners
+  */
+  registerRpcListeners() {
+    this.rpc.on('socket-error', err => this.emit('socket-error', err));
+    this.rpc.on('closed', (evt) => {
+      if ((this.suspendOnClose && evt.code !== 1000) || evt.code === MANUAL_SUSPEND) {
+        this.emit('suspended',
+          { initiator: evt.code === MANUAL_SUSPEND ? 'manual' : 'network' });
+      } else {
+        this.emit('closed', evt);
+      }
+    });
+
+    this.rpc.on('message', (response) => {
+      if (response.change) {
+        response.change.forEach(handle => this.emit('handle-changed', handle));
+      }
+
+      if (response.close) {
+        response.close.forEach(handle => this.emit('handle-closed', handle));
+      }
+    });
+  }
+
+  // TODO: Rename as this does not unregister the notification listerners.
+  /**
+  * Function used unregister the RPC listerners except for the notification listeners
+  */
+  unregisterRpcListeners() {
+    this.rpc.removeAllListeners('socket-error');
```

Comment:
Nitpicking, but the session should probably not remove _all_ listeners on the RPC instance (just the ones it owns itself), if we add functionality in the future that binds events on the RPC instance they might subtly get removed here, causing issues.